src/component.rs

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2020 Red Hat, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use anyhow::Result;
+use std::fs::File;
+use std::io::Write as IoWrite;
+use std::path::{Path, PathBuf};
+
+use crate::model::*;
```

Comment:
Here and everywhere else, if you could avoid star-imports the code could be a bit easier to read/navigate.